service_container.rst

```diff
@@ -690,6 +690,40 @@ argument for *any* service defined in this file! You can bind arguments by name
 The ``bind`` config can be also be applied to specific services or when loading many
 services at once (i.e. :ref:`service-psr4-loader`).
 
+Getting Container Parameters as a Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.1
+    The ``ContainerBagInterface`` class was introduced in Symfony 4.1.
+
+If some service or controller needs lots of container parameters, there's an
+easier alternative to binding all of them. Type-hint an argument with the
+:class:`Symfony\\Component\\DependencyInjection\\ParameterBag\\ContainerBagInterface`
+class to inject all container parameters in a single object similar to the
+:class:`Symfony\\Component\\DependencyInjection\\ParameterBag\\ParameterBag`
+objects used in other parts of the framework::
+
+    // src/Service/MessageGenerator.php
+    // ...
+
+    use Symfony\Component\DependencyInjection\ParameterBag\ContainerBagInterface;
+
+    class MessageGenerator
```

Comment:
Maybe a more concrete example with a Mailer class using a `mailer_sender` parameter in a `sendWelcomeEmail()` method maybe?

Reply:
I like the idea of giving a name to the parameter. Thanks!